Salehi2025/Figure5.qmd

```diff
@@ -0,0 +1,450 @@
+---
+title: "Figure5 Data Processing and Visualization"
+format: html
+editor: visual
+---
+
+# Indroduction
+
+This document contains the data processing, analysis, and visualization steps for reproducing Figure 5. The statistical models used in this analysis were precomputed in a separate Quarto document (Figure5models.qmd). It can be found from [GitHub](https://github.com/microbiome/data/blob/main/Salehi2025). This separation ensures that model fitting, which is computationally intensive, does not need to be rerun every time the report is generated.
+
+Let's load all the necessary libraries for data processing, modeling, and visualization.
+
+```{r setup, include=FALSE}
+library(scater)
+library(ggpubr)
+library(rstatix)
+library(viridis)
+library(ggsignif)
+library(Matrix)
+library(RColorBrewer)
+library(caret)
+library(randomForest)
+library(xgboost)
+library(brms)
+library(glmnet)
+library(tidyverse)
+library(vegan)
+library(TreeSummarizedExperiment)
+```
+
+The dataset used in this analysis, tse_AMRdemo.rds, is available in the [GitHub repository](https://github.com/microbiome/data/blob/main/Salehi2025/tse_AMRdemo.rds)
+
+```{r}
+# Load TSE object
+tse <- tse_AMRdemo
+```
+
+# Data processing
+
+Now we filter out samples with missing gender or age information, extracts relevant metadata, computes total read counts per sample, and applies a log-transformation to the ARG load variable. These steps prepare the dataset for downstream analysis and modeling.
+
+```{r, Readcount}
+# Filter samples with complete gender and age data
+non_na_samples <- !is.na(colData(tse)$gender) & !is.na(colData(tse)$host_age_years)
+TSE_filtered <- tse[, non_na_samples]
+
+# Extract metadata
+adult_metadata <- as.data.frame(colData(TSE_filtered))
+
+# Readcount
+counts_mat <- assay(TSE_filtered, "counts")
+adult_metadata$readcount_2 <- colSums(counts_mat)
+
+adult_metadata$log_ARG_load <- log(adult_metadata$ARG_load)

```

Comment:
check if we really need the separate variable for log_ARG_load or can we deal with just ARG_load (and call log-normal models etc as needed)